lib/translator/locale-parser.js

```diff
@@ -37,10 +37,17 @@ function getPreferredLocale(acceptLanguage, languages, defaultLocale) {
 function getLocales(acceptLanguage) {
     const localeObjects = AcceptLanguageParser.parse(acceptLanguage);
 
-    return _.map(localeObjects, localeObject => {
+    const localesArr = _.map(localeObjects, localeObject => {
         return _.isString(localeObject.region) ? `${localeObject.code}-${localeObject.region}` : localeObject.code;
     });
-}
+
+    //  Safari sends only one language code, this is to have a default fallback in case we don't have that language
+    //  As an example we may not have fr-FR so add fr to the header
+    if (localesArr.length === 1 && localesArr[0].split('-').length === 2) {
```

Comment:
How does this work? I thought `localesArr[0]` would be an object, not a string.

Reply:
localesArr[0] would be an object after acceptlanguageparser.parse but I'm doing this after the map function which makes things into an array of strings.